packaging/pep517_backend/_backend.py

```diff
@@ -378,8 +374,7 @@ def get_requires_for_build_wheel(
         )
 
     c_ext_build_deps = [] if is_pure_python_build else [
-        'Cython >= 3.0.0b3' if IS_PY3_12_PLUS  # Only Cython 3+ is compatible
-        else 'Cython',
+        'Cython >= 3.0.0, < 3.1.0',
```

Comment:
Let's not put the lower limit for older CPython versions.
```suggestion
        'Cython ~= 3.0.0; python_version >= "3.12"',
        'Cython; python_version < "3.12"',
```